sematic/ui/src/pipelines/external_resource/ExternalResourceState.tsx

```diff
@@ -0,0 +1,28 @@
+import TimelineConnector from "@mui/lab/TimelineConnector/TimelineConnector";
+import TimelineContent from "@mui/lab/TimelineContent/TimelineContent";
+import TimelineDot from "@mui/lab/TimelineDot/TimelineDot";
+import TimelineItem from "@mui/lab/TimelineItem/TimelineItem";
+import TimelineSeparator from "@mui/lab/TimelineSeparator/TimelineSeparator";
+import { HistorySerializationType, TERMINATE_STATE } from "../../hooks/externalResourceHooks";
+import { format } from 'date-fns'
+import { useMemo } from "react";
+import Typography from "@mui/material/Typography/Typography";
+
+export default function ExternalResourceState({state}: {state: HistorySerializationType}) {
+    const {id, status_message, created_at} = state;
+    const timeString = useMemo(
+        () => format(
+            created_at instanceof Date ? created_at : new Date(created_at), 
+            'LLL\xa0d,\xa0yyyy\xa0h:mm:ss\xa0a'), [created_at]);
+
+    return <TimelineItem key={id}>
+        <TimelineSeparator>
+            <TimelineDot />
+            {status_message !== TERMINATE_STATE && <TimelineConnector />}
+        </TimelineSeparator>
+        <TimelineContent>
+            {status_message}
+            <Typography>{timeString}</Typography>
+        </TimelineContent>
```

Comment:
We also need to display the resource name.

Reply:
Sounds good. It might be easier for me to implement this when the server side is ready so I can try out and see what the payload looks like.

Reply:
Added resource name in the latest iteration.